lib/matplotlib/tests/test_image.py

```diff
@@ -1606,3 +1642,84 @@ def test_non_transdata_image_does_not_touch_aspect():
     assert ax.get_aspect() == 1
     ax.imshow(im, transform=ax.transAxes, aspect=2)
     assert ax.get_aspect() == 2
+
+
+@image_comparison(
+    ['downsampling.png'], style='mpl20', remove_text=True, tol=0.09)
+def test_downsampling():
+    N = 450
+    x = np.arange(N) / N - 0.5
+    y = np.arange(N) / N - 0.5
+    aa = np.ones((N, N))
+    aa[::2, :] = -1
+
+    X, Y = np.meshgrid(x, y)
+    R = np.sqrt(X**2 + Y**2)
+    f0 = 5
+    k = 100
+    a = np.sin(np.pi * 2 * (f0 * R + k * R**2 / 2))
+    # make the left hand side of this
+    a[:int(N / 2), :][R[:int(N / 2), :] < 0.4] = -1
+    a[:int(N / 2), :][R[:int(N / 2), :] < 0.3] = 1
+    aa[:, int(N / 3):] = a[:, int(N / 3):]
+    a = aa
+
+    fig, axs = plt.subplots(2, 3, figsize=(7, 6), layout='compressed')
+    axs[0, 0].imshow(a, interpolation='nearest', interpolation_stage='rgba',
+                     cmap='RdBu_r')
+    axs[0, 0].set_xlim(125, 175)
+    axs[0, 0].set_ylim(250, 200)
+    axs[0, 0].set_title('Zoom')
+
+    for ax, interp, space in zip(axs.flat[1:], ['nearest', 'nearest', 'hanning',
+                                                'hanning', 'antialiased'],
+                                 ['data', 'rgba', 'data', 'rgba', 'antialiased']):
+        ax.imshow(a, interpolation=interp, interpolation_stage=space,
+                  cmap='RdBu_r')
+        ax.set_title(f"interpolation='{interp}'\nspace='{space}'")
+
+
+@image_comparison(
+    ['downsampling_speckle.png'], style='mpl20', remove_text=True, tol=0.09)
+def test_downsampling_speckle():
+    fig, axs = plt.subplots(1, 2, figsize=(5, 2.7), sharex=True, sharey=True,
+                            layout="compressed")
+    axs = axs.flatten()
+    img = ((np.arange(1024).reshape(-1, 1) * np.ones(720)) // 50).T
+
+    cm = plt.get_cmap("viridis")
+    cm.set_over("m")
+    norm = colors.LogNorm(vmin=3, vmax=11)
+
+    # old default cannot be tested because it creates over/under speckles
+    # in the following that are machine dependent.
+
+    axs[0].set_title("interpolation='antialiased', stage='rgba'")
+    axs[0].imshow(np.triu(img), cmap=cm, norm=norm, interpolation_stage='rgba')
+
+    # Should be same as previous
```

Comment:
Can you do a figure reference/test comparison test here instead with these two axes?